packages/react-instantsearch-hooks/src/useIndex.ts

```diff
@@ -1,30 +1,60 @@
 import index from 'instantsearch.js/es/widgets/index/index';
-import { useEffect, useMemo } from 'react';
+import { useEffect, useMemo, useRef } from 'react';
 
 import { useForceUpdate } from './useForceUpdate';
 import { useIndexContext } from './useIndexContext';
+import { useInstantSearchServerContext } from './useInstantSearchServerContext';
+import { useInstantSearchSSRContext } from './useInstantSearchSSRContext';
 import { useStableValue } from './useStableValue';
+import { useIsomorphicLayoutEffect } from './utils';
 
 import type { IndexWidgetParams } from 'instantsearch.js/es/widgets/index/index';
 
 export type UseIndexProps = IndexWidgetParams;
 
 export function useIndex(props: UseIndexProps) {
+  const serverContext = useInstantSearchServerContext();
+  const ssrContext = useInstantSearchSSRContext();
   const parentIndex = useIndexContext();
   const stableProps = useStableValue(props);
-  const indexWidget = useMemo(() => index(stableProps), [stableProps]);
+  // On SSR rendering, we add the Index early in a memo to render the initial
+  // results in the render pass. Since the Index is added already, we need to
+  // skip the usual browser effect that adds the widget when client-side
+  // rendering. We still need to run the effect lifecycle on all the other renders.
+  // This ref lets us keep track whether the initial effect to add the Index
```

Comment:
```suggestion
  // This ref lets us track whether the initial effect to add the Index
```

Reply:
The original sentence is also correct, no? That's fewer words.

Reply:
My suggestion was missing a word, it should have been "keep track **of**".

I'm fine with your current sentence.